packages/core/test/polygon/e2e.js

```diff
@@ -0,0 +1,176 @@
+const { ZERO_ADDRESS, interfaceName, RegistryRolesEnum, didContractThrow } = require("@uma/common");
+const { assert } = require("chai");
+const TruffleAssert = require("truffle-assertions");
+
+const { utf8ToHex, toWei, hexToUtf8 } = web3.utils;
+
+const StateSync = artifacts.require("StateSyncMock");
+const FxChild = artifacts.require("FxChildMock");
+const FxRoot = artifacts.require("FxRootMock");
+const OracleChildTunnel = artifacts.require("OracleChildTunnel");
+const OracleRootTunnel = artifacts.require("OracleRootTunnelMock");
```

Comment:
Why are we using a mock for the root tunnel but the real contract for the child tunnel? Just curious, not suggesting we shouldn't!

Reply:
Ah I'll add a comment. Because the `RootTunnel` has an internal `_processMessageFromChild` that we want to test directly, whereas the corresponding `_processMessageFromRoot` on the `ChildTunnel` gets tested via `FxChildMock.onStateReceive`.